CliWrap/Utils/ProcessEx.cs

```diff
@@ -62,12 +65,17 @@ public void Start()
         };
 
         // Start the process
-        if (!_nativeProcess.Start())
+        try
         {
-            throw new InvalidOperationException(
-                "Failed to obtain the handle when starting a process. " +
-                "This could mean that the target executable doesn't exist or that execute permission is missing."
-            );
+            _nativeProcess.Start();
+        }
+        catch (Exception ex)
+        {             
+            var message = $"Fail to execute '{_fileName}'.\n";
+            if (ex is System.ComponentModel.Win32Exception w && w.NativeErrorCode == 2)
+                message += "This could mean that the target executable doesn't exist or that execute permission is missing.\n";
+            message += "-----The original thrown Exception-----\n" + ex.ToString();
```

Comment:
No need for "The original thrown Exception", just add it as `innerException` in below constructor and it will be included in the message automatically.